articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,112 @@ A random delay is added before the cached value is marked as dirty to reduce pot
+```
+This method is called by the refresher and provider to process the `PushNotification`. `ProcessPushNotification()` will invoke `UpdateSyncToken()` and then invoke `SetDirty()` to ensure the most recent configuration is provided to the user.
+
+Open **Program.cs** and update the file with the following code.
+This example uses [ServiceBus](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dotnet-get-started-with-queues) as an example. 
+
+```cs
+using Azure.Messaging.EventGrid;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using System;
+using System.Threading.Tasks;
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Examples.ConsoleApplication
+{
+    class Program
+    {
+        private const string AppConfigurationConnectionStringEnvVarName = "AppConfigurationConnectionString"; // e.g. Endpoint=https://{store_name}.azconfig.io;Id={id};Secret={secret}
+        private const string ServiceBusConnectionStringEnvVarName = "ServiceBusConnectionString"; // e.g. Endpoint=sb://{service_bus_name}.servicebus.windows.net/;SharedAccessKeyName={key_name};SharedAccessKey={key}
+        private const string ServiceBusTopicEnvVarName = "ServiceBusTopic";
+        private const string ServiceBusSubscriptionEnvVarName = "ServiceBusSubscription";
+        private static IConfigurationRefresher _refresher = null;
+        static async Task Main(string[] args)
+        {
+            string appConfigurationConnectionString = Environment.GetEnvironmentVariable(AppConfigurationConnectionStringEnvVarName);
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(appConfigurationConnectionString);
+                    options.ConfigureRefresh(refresh =>
+                        refresh
+                            .Register("TestApp:Settings:Message")
+                            .SetCacheExpiration(TimeSpan.FromDays(30))  // Important: Reduce poll frequency
+                    );
+                    _refresher = options.GetRefresher();
+                }).Build();
+            RegisterRefreshEventHandler();
+            var message = configuration["TestApp:Settings:Message"];
+            Console.WriteLine($"Initial value: {configuration["TestApp:Settings:Message"]}");
+            while (true)
+            {
+                await _refresher.TryRefreshAsync();
+                if (configuration["TestApp:Settings:Message"] != message)
+                {
+                    Console.WriteLine($"New value: {configuration["TestApp:Settings:Message"]}");
+                    message = configuration["TestApp:Settings:Message"];
+                }
+                await Task.Delay(TimeSpan.FromSeconds(1));
+            }
+        }
+        private static void RegisterRefreshEventHandler()
```

Comment:
We dont need the entire code again. We can ask users to add the new package dependency for Azure.Messaging.EventGrid and show the changes in `RegisterRefreshEventHandler` method. Everything else is same as the snippet in the previous section.